### PR TITLE
Remove Cython APIs for table view generation

### PR DESCRIPTION
**Pull request by vyasr**
This PR removes the Table methods used to generate `cudf::table_view` objects in Cython and switches all code paths to use the underlying `cdef` function that those functions originally called under the hood. Many of the view-generating functions were actually never used (all the mutable view generators and `index_view`), so those have been removed. Moving in this direction has two benefits. First, it allows us to change our Cython APIs to accept iterables of columns rather than `Table` objects, which in turn means that we can get rid of various Column->Table->Column round-trips that are currently in place just to call Cython APIs on Columns that are only defined on Tables. Second, the Table class is now pure Python, so its logic can be merged into the Frame class.

### CODE REVIEW

**Comment by vyasr** [comment]
This only affects packages relying on our Cython directly rather than using public Python APIs, but just in case CCing @rlratzel for cuGraph, @dantegd for cuML, and @thomcom for cuSpatial.

File: python/cudf/cudf/_lib/concat.pyx
Comment (isVoid) [inline python/cudf/cudf/_lib/concat.pyx:47]: We can use `utils.pyx/make_table_views()`

File: python/cudf/cudf/_lib/sort.pyx
Comment (isVoid) [inline python/cudf/cudf/_lib/sort.pyx:230]: From readability perspective, I think we should explicitly use keyword argument when there's a lone boolean value:
```suggestion
    cdef table_view source_table_view = table_view_from_table(
        source_table, ignore_index=True
    )
```

File: python/cudf/cudf/_lib/interop.pyx
Comment (isVoid) [inline python/cudf/cudf/_lib/interop.pyx:67]: ditto

File: python/cudf/cudf/_lib/strings/combine.pyx
Comment (isVoid) [inline python/cudf/cudf/_lib/strings/combine.pyx:36]: ditto

File: python/cudf/cudf/_lib/transform.pyx
Comment (isVoid) [inline python/cudf/cudf/_lib/transform.pyx:130]: ditto

File: python/cudf/cudf/_lib/transform.pyx
Comment (isVoid) [inline python/cudf/cudf/_lib/transform.pyx:153]: ditto

File: python/cudf/cudf/_lib/transpose.pyx
Comment (isVoid) [inline python/cudf/cudf/_lib/transpose.pyx:49]: ditto

**Review by isVoid — COMMENTED** [review]
Same issue as above.

**Comment by vyasr** [comment]
@isVoid OK I went through and checked all the files. I think I got all the places where I was passing through a raw boolean.

**Review by isVoid — COMMENTED** [review]
Thanks, this mostly looks good from my end. But I think we should wait until the maintainers mentioned above comment on this.

**Comment by vyasr** [comment]
> Thanks, this mostly looks good from my end. But I think we should wait until the maintainers mentioned above comment on this.

Agreed, I don't plan to merge until other maintainers can take a look.

**Comment by vyasr** [comment]
Got the OK on this from @dantegd offline

**Comment by rlratzel** [comment]
I re-checked cuGraph and didn't see anything that's using these Cython APIs, so this change should be safe for cuGraph.  I believe we're only using public cuDF python APIs in the pure python layer.

**Comment by vyasr** [comment]
OK great. I've opened rapidsai/cuspatial#449 to make cuspatial work after this PR, so we should be good to go once this PR is merged.

**Review by isVoid — APPROVED** [review]

**Review by galipremsagar — APPROVED** [review]

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**
